geode-apis-compatible-with-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/server/CommandIntegrationTest.java

```diff
@@ -73,6 +76,25 @@ public void commandReturnsResultsMatchingNativeRedis() {
     softly.assertAll();
   }
 
+  @Test
+  public void commandDoesNotReturnUnsupported_whenUnsupportedCommandsAreDisabled() {
+    radishServer.setEnableUnsupportedCommands(false);
+    try {
+      Map<String, CommandStructure> results = processRawCommands(radishClient.command());
+
+      // Find an unsupported command
+      RedisCommandType someUnsupported = Arrays.stream(RedisCommandType.values())
+          .filter(RedisCommandType::isUnsupported).findFirst()
+          .orElseThrow(() -> new AssertionFailedError("Could not find any UNSUPPORTED commands"));
+
+      for (CommandStructure meta : results.values()) {
+        assertThat(meta.name).isNotEqualToIgnoringCase(someUnsupported.name());
+      }
+    } finally {
+      radishServer.setEnableUnsupportedCommands(true);
+    }
```

Comment:
It's generally best and cleaner if you can avoid using finally-blocks for tear down. You should consider hoisting the finally-block to a dedicated tear down method instead:
```
@After
public void tearDown() {
  radishServer.setEnableUnsupportedCommands(true);
}
```
You can always add a conditional to the tear down if needed.

Reply:
Thanks - I've updated per your suggestion.